### ISSUE TITLE
Fix file picker which is broken in Thunderbird 125

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#476]
According to bug [1878401](https://bugzilla.mozilla.org/show_bug.cgi?id=1878401) a parameter was changed in mozilla-central for the interface `nsIFilePicker.init()` - it will fail when we pass in `window` as first parameter. The c-c source code shows that instead we need to plug in `window.browsingContext`. This needs to be fixed in a way so that QuickFolders stays compatible with Thunderbird 115 but is also compatible with Thunderbird 125 and later.

**Comment by RealRaven2000** [comment]
Here is what happens (tested in Tb 128.0b1) when we try to save the configuration from the license tab:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/912cd602-4306-4e0b-a8d0-b69ff94a6283)

no file picker dialog appears; in error console we see the following exception: 

```log
[Exception... "Could not convert JavaScript argument arg 0 [nsIFilePicker.init]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: jar:file:///I:/Thunderbird_Profiles/ct4akusa.Beta-Test/extensions/quickfolders@curious.be.xpi!/chrome/content/api/Utilities/implementation.js :: fileConfig :: line 35"  data: no]

```

**Comment by RealRaven2000** [comment]
Here is a fixed version:

[QuickFolders-mx-6.5.2pre38.zip](https://github.com/user-attachments/files/15968613/QuickFolders-mx-6.5.2pre38.zip)

This should also be backwards compatible with ESR 115 and all versions in between that and 128.* - no new branch is planned!

----------

To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by RealRaven2000** [comment]
Fixed in 6.6 - Published 21/07/2024
